### ISSUE TITLE
Lag on laptop with Ivy Bridge Core i3 and 4GB RAM.

### ISSUE THREAD

**Issue by groudie** [jderose9/dash-to-panel#65]
At least I am assuming that's why I get the lag. The machine is an Asus x202e(specs can be found here: https://www.cnet.com/products/asus-vivobook-x202e-dh31t/specs/) It doesn't show up on my dell with 16GB RAM and a Haswell Core i7. I notice the lag happening consistently in two instances:

1. When coming out of Window Spread mode.
2. When clicking on an app in the dock the app icons animation is also laggy. 

I don't get this issue on dash-to-dock or with the default gnome desktop dock setup.

What I've tried to solve the problem:
1. Disable 'animate show applications'
2. Try both the AUR and gnome-shell-extensions install route.
3. Used it with impatience(altering the speed of animation to see if that helped) and without impatience.

The only time it doesn't lag is when I disable animations in the Application section of the Gnome Tweak Tool.

**Comment by jderose9** [comment]
Hi there! Thanks for reporting this. Do you think this is the same issue described in #44?

**Comment by groudie** [comment]
Thanks for answering. 

Hmmm, I think my problem is similar yet different. The only time I notice the lag is when coming out of window spread. The lag is very noticeable when the different windows that were spread are converging and going back to their original position on the desktop, and also when the virtual desktop selector on the right slides back out of view. When clicking on an application icon there is an animation where the icon gets large and fades out of view quickly. That too is laggy. I can't remember it lagging when opening lists menus though. I'll have to check when I get back to work tomorrow. 

The setups on both machine are basically the same - Antergos with Gnome 3.22.2. I didn't try disabling all other extensions though. For example 'Dynamic Panel Transparency' and 'Status Menu Buttons' were enabled. I'll disable them all and leave only dash to panel and report back.

**Comment by groudie** [comment]
Reporting back.

I noticed 2 things in particular.

1. There is like when opening drop down menus.

2. The lag **does not** show when I disable the 'dynamic panel transparency' extension. 

Disabling anything else will not resolve the issue I can confirm that using dash-to-dock and dynamic panel transparency doesn't cause the same issues.

**Comment by jderose9** [comment]
Thanks for the additional info. What do your CPU and RAM usage look like when this is occurring?

Is there any way you could make a video capture or recording of this?

**Comment by lnicola** [comment]
This extension is very sluggish for me when one of the "drop-down" menus is open and I move the mouse over the right side of the panel. I'm on Wayland with Sandy Bridge integrated graphics.

When this happens, GNOME Shell uses 50% CPU, mostly in `pixman_region32_translate`. Could this be the same issue?

I'm referring to this area of the panel:
![image](https://cloud.githubusercontent.com/assets/308347/22926631/1851b2f2-f2b6-11e6-8f4b-483fbfd976a0.png)

**Comment by groudie** [comment]
I documented the issue [here](https://vimeo.com/204001542) 

The video tearing was caused by the screen recorder and has nothing to do with panel-to-dash. I started off with dynamic transparency panel disabled and then enabled it to show the difference. I forgot to document the CPU and RAM usage in the video but from what I am seeing here there doesn't seem to be a noticeable difference in RAM and CPU usage when dynamic transparency panel is enabled.

**Comment by jderose9** [comment]
@lnicola How were you able to narrow down to that function? Did you attach a debugger to your compositor? Your issue sounds more like that described in #44. I'm curious if setting all the sliders in the Appearance tab to the default (all the way to the left) still produces this issue? And it definitely doesn't occur with this extension disabled?

@groudie Thanks a lot for the video. It seems like this issue occurs when you have multiple animations simultaneously? for instance, when you open a first maximized window, the panel is darkened at the same time the launch animation occurs. If you launch a 2nd application with the panel already darkened, does the same issue occur?

**Comment by lnicola** [comment]
@jderose9 With `perf top`, but now I get different functions at the top of the profile, so I'm not sure what to make of it.

Do you mean these sliders?
![image](https://cloud.githubusercontent.com/assets/308347/22937079/2034a126-f2e0-11e6-9ef7-fa73fef4ec19.png)

Behavior tab:
![image](https://cloud.githubusercontent.com/assets/308347/22937088/2a19731a-f2e0-11e6-84c7-94a30519c1c4.png)

But you're right, the high CPU usage also occurs without this extension, so please ignore my post. I think it's more noticeable with Dash to panel because of the blue highlight which shows exactly where it's registering the mouse. Without the highlight one can assume it's an animation or menu open delay.

**Comment by groudie** [comment]
@jderose9 No problem.

I've tried launching a second application with the panel darkened and the same issue occurs. 

With the Dynamic transparency panel I've set the panel to be transparent except when an app window is maximized. When the window is maximized the panel becomes opaque after a slight delay(I've purposely set it that way in the dtp settings). You think this is causing the issue? When I use dash-to-dock the same thing happens except that the panel is at the top. For some reason the lag doesn't show when I use dash-to-dock and dtp together.

**Comment by jderose9** [comment]
@lnicola Yes, I was speculating that if one of the widgets was re-adding itself to the panel then it could potentially cause redraws as the styles were set again. That code doesn't execute if the sliders are all set to the left. But if it happens without this extension installed, that definitely isn't the issue.

@groudie When you are using dash-to-dock, you have it set to always be visible? And the application launch animation occurs with dash-to-dock too?

**Comment by groudie** [comment]
@jderose9 Yes to both of those questions

**Comment by jderose9** [comment]
@groudie I've made a couple of changes that I'm hoping will improve this situation. Can you try the latest code from the `master` branch?

The first thing is that I found several scenarios where the shell was repainting repeatedly and the animations were being trigger at the same time or redundantly and I've tried to work around those. So if you just try the new code it may be better for you.

If that doesn't work, I added two new settings to remove animation when switching apps and when launching apps. They are on the "Fine-Tune" tab of the settings. If you could try to shut those off and see if your behavior is any better?

**Comment by groudie** [comment]
@jderose9 thanks for the response and the hard work. I ran the following in Terminix: 

> git clone --depth 1 https://github.com/jderose9/dash-to-panel.git
> 
> cd dash-to-panel
> 
> make
> 
> make install

Unfortunately the problem is still there even when I switch off 'Animate switch applications' and 'Animate launching new windows'.

**Comment by groudie** [comment]
@jderose9 Hello again. I have an update. After doing a fresh install of Antergos the problem seems to have disappeared. I am no longer seeing any lag at all when Dynamic Transparency Panel is enabled. That was with the version that is currently on Gnome Shell Extensions website. I'll keep monitoring it for updates.

**Comment by jderose9** [comment]
I'm glad you got this issue resolved! Thanks for reporting back. Since this is confirmed as the same issue as the other thread and you seem to be in good shape, I'm going to close this one for now. Definitely let me know if it re-occurs. Thanks for your help!

**Comment by groudie** [comment]
@jderose9 I think I can confirm now that your extension may not have been the problem. Remember I mentioned that the issue only shows up when dynamic transparency panel and dash to panel are enabled together? Well I did a fresh install of Manjaro on my dell and the lag is there when when I have dynamic transparency panel enabled. It's there even when I don't have dash to panel installed. 

I just thought I should let you know. I've already made a bug report on the issue.
